proxy/hdrs/MIME.h

```diff
@@ -1304,10 +1303,8 @@ inline int
 MIMEHdr::value_get_index(const char *name, int name_length, const char *value, int value_length) const
 {
   const MIMEField *field = field_find(name, name_length);
-  if (field)
-    return field->value_get_index(value, value_length);
-  else
-    return -1;
+
+  return field ? field->value_get_index(value, value_length) : -1;
```

Comment:
This doesn't look great to me. I'd like to keep it as is, we should add add "{" and "}" to the if statement in that case though.

Another option would be using C++17 style.
```
if (auto field = field_find(name, name_length); field) {
  return field->value_get_index(value, value_length);
} else {
  return -1
}
```

Reply:
Hmmm, really ? I find the tertiary construct easy to read. Anyone else have input on this ?

Reply:
I don't think this line is ugly, but if we made a change on this line, the diff would not be so clear. Imagine a case adding a debug log. You would need to convert it to `if` statement.

Reply:
Sure, but that's future / forward thinking, no? The same could be said for any tertiary expression, and with that logic, you'd want to ban them all? :-) These getters are unlikely to add Debug() IMO.

Reply:
My personal preference is to not use ternary expressions at all actually. The cons of ternary expressions I think are:
- Unclear diffs on changes
- Inability to set a breakpoint on either true or false branch
- More possibility of merge conflicts

In my relatively short life, I've never seen a ternary expression that does a great job defeating the cons above. There may be special cases, but I don't think this is one of them.

Reply:
Why was cppcheck saying there was an error in the first place?  I agree with @maskit points on the downsides of the ternary operator.  I am not a big fan of it and rarely use it myself.

Reply:
I don't know that it complained about all of them, but it's about where you declare the local variables or not. I can undo the ternary stuff.